src/nncf/quantization/algorithms/weight_compression/awq.py

```diff
@@ -248,6 +263,20 @@ def _data_aware_step(self, wp, weight, statistics):
             alpha = self._alpha_min
             for _ in range(self._steps):
                 cur_scale = gscale**alpha
+
+                if prev_s is not None:
+                    threshold = 0.25 * 65504  # take the threshold from the fp16 type with some margin
```

Comment:
As I understand 65504 is selected assuming inference is performed in FP16. Won't it negatively affect FP32 inference? And BF16 (SPR) case should be considered too. Can you please provide validation metrics for these three scenarios and a baseline?